src/Domain/SegmentId.hpp

```diff
@@ -66,34 +68,33 @@ bool operator!=(const SegmentId& lhs, const SegmentId& rhs) noexcept;
 // INLINE DEFINITIONS
 //##############################################################################
 
-inline SegmentId SegmentId::id_of_parent() const {
-  ASSERT(0 != refinement_level_,
-         "Cannot call id_of_parent() on root refinement level!");
+inline SegmentId SegmentId::id_of_parent() const noexcept {
+  CASSERT(0 != refinement_level_,
+          "Cannot call id_of_parent() on root refinement level!");
   // right-shifting index_ by 1 gives us floor(index_/2).
   // The parent has half as many segments as the child.
-  return SegmentId(refinement_level_ - 1, index_ >> 1);
+  return {refinement_level_ - 1, index_ >> 1};
```

Comment:
Was this bit magic used here to solve performance problems? If not, I'd favor the much more readable `floor(index_/2)` given in the comment.